converter/solid_figures.py

```diff
@@ -0,0 +1,75 @@
+from dataclasses import dataclass
+from abc import ABC
+
+
+@dataclass(frozen=True)
+class SolidFigure(ABC):
+    """Abstract figure, bodies are used to define geometries."""
+
+    uuid: str
+    offset: list[float]
+    rotation: list[float]
+
+
+@dataclass(frozen=True)
+class SphereFigure(SolidFigure):
+    """
+    A sphere figure. It's defined by its radius, offset from the point (0, 0, 0) xyz
+    and its rotation along each axis.
+    """
+
+    radius: float
+
+
+@dataclass(frozen=True)
+class CylinderFigure(SolidFigure):
+    """
+    A cylinder figure. It's defined by the radii of both of its bases(top and bottom),
+    its height, its offset from the point (0, 0, 0) xyz and its rotation along each axis.
+    """
+
+    radius_top: float
+    radius_bottom: float
+    height: float
+
+
+@dataclass(frozen=True)
+class BoxFigure(SolidFigure):
+    """
+    A box figure. It's defined by size in the 3 dimensions (z - height, x - width and y - depth),
+    its offset from the point (0, 0, 0) xyz and its rotation along each axis.
+    """
+
+    height: float
```

Comment:
`height`, `width` and `depth` do not carry information about the axis (X,Y,Z) along which the distance is measured.
I'd suggest `x_edge_length` or `length_x` or `x_length` or `x_size` etc...

Reply:
Since the object can be rotated i don't think we should correlate its dimensions with X,Y,Z axes.  Maybe we could put a disambiguation in the docstring explaining what each dimension means when the rotation is (0, 0, 0)?

Reply:
But the size is defined in local coordinate systems. Users are very much used to it.
See manuals for SH12A or TOPAS

Reply:
Changed to the `x_edge_length` convention since it strongly suggests that the measure concerns the edges of the figure.

Reply:
ok